source/common/umf_pools/disjoint_pool_config_parser.hpp

```diff
@@ -54,6 +72,30 @@ class DisjointPoolAllConfigs {
 // "1;32M;host:1M,4,64K;device:1M,4,64K;shared:0,0,2M"
 DisjointPoolAllConfigs parseDisjointPoolConfig(const std::string &config,
                                                int trace = 1);
+
+static void UMF_CALL_THROWS(umf_result_t res) {
+    if (res != UMF_RESULT_SUCCESS) {
+        throw res;
+    }
+}
+
+static inline umfDisjointPoolParamsHandle
+getUmfParamsHandle(umf_disjoint_pool_config_t &config) {
+    umf_disjoint_pool_params_handle_t handle;
+    UMF_CALL_THROWS(umfDisjointPoolParamsCreate(&handle));
+    UMF_CALL_THROWS(
+        umfDisjointPoolParamsSetSlabMinSize(handle, config.SlabMinSize));
+    UMF_CALL_THROWS(umfDisjointPoolParamsSetMaxPoolableSize(
```

Comment:
If some `umfDisjointPoolParamsSet*` call fails, the `UMF_CALL_THROWS` will throw an exception, but the `handle` object is not destructed, as a result, there is a memory leak because the `umfDisjointPoolParamsDestroy` is not called.

Reply:
Of course, done.